### PR TITLE
Added ability to create image from archive

### PR DESCRIPTION
**Pull request by diptanu**
This PR adds the ability to load an image from an archive file.

The Driver Config supports a new key called `load` where the name of the archive has to be mentioned.

Users can use the `artifact` block in a Task to download the archive file, and Nomad looks for the archive name mentioned in the `load` key in the `/taskdir/local` directory.

### CODE REVIEW

File: client/driver/docker.go [hunk 1]
```diff
@@ -470,63 +440,117 @@ func (d *DockerDriver) Start(ctx *ExecContext, task *structs.Task) (DriverHandle
...
 
-		// Now that we have the image we can get the image id
-		dockerImage, err = client.InspectImage(image)
-		if err != nil {
-			d.logger.Printf("[ERR] driver.docker: failed getting image id for %s: %s", image, err)
-			return nil, fmt.Errorf("Failed to determine image id for `%s`: %s", image, err)
-		}
+// loadImage creates an image by loading it from the file system
+func (d *DockerDriver) loadImage(driverConfig *DockerDriverConfig, client *docker.Client, taskDir string) error {
+	archive := filepath.Join(taskDir, allocdir.TaskLocal, driverConfig.LoadImage)
```
Comment (dadgar) [inline client/driver/docker.go]: I am not sure you want to do that because artifacts can be downloaded to arbitrary locations

File: client/driver/docker.go [hunk 2]
```diff
@@ -470,63 +440,117 @@ func (d *DockerDriver) Start(ctx *ExecContext, task *structs.Task) (DriverHandle
...
 	if dockerImage == nil {
-		pullOptions := docker.PullImageOptions{
-			Repository: repo,
-			Tag:        tag,
+		if driverConfig.LoadImage != "" {
+			return d.loadImage(driverConfig, client, taskDir)
+		} else if driverConfig.ImageName != "" {
+			return d.pullImage(driverConfig, client, repo, tag)
+		} else {
+			return fmt.Errorf("either image name or load image has to be specified")
```
Comment (dadgar) [inline client/driver/docker.go]: Probably want to put a check in the validate that if load image is specified so is image name

File: client/driver/docker.go [hunk 3]
```diff
@@ -470,63 +440,117 @@ func (d *DockerDriver) Start(ctx *ExecContext, task *structs.Task) (DriverHandle
...
+	var driverConfig DockerDriverConfig
+	if err := mapstructure.WeakDecode(task.Config, &driverConfig); err != nil {
+		return nil, err
+	}
+
+	if err := driverConfig.Init(); err != nil {
+		return nil, err
 	}
 
+	image := driverConfig.ImageName
```
Comment (dadgar) [inline client/driver/docker.go]: Can you move this down to where it is first used

File: client/network/interfaces.go [hunk 4]
```diff
@@ -0,0 +1,6 @@
+package network
+
+import ()
+
```
Comment (dadgar) [inline client/network/interfaces.go]: Why is this change here?

**Comment by dadgar** [comment]
Website documentation as well.

File: client/driver/docker.go (hunk 2 above)
Comment (diptanu) [inline client/driver/docker.go]: @dadgar `Validate` already ensures that Image Name is specified in all cases.
